idaes_examples/notebooks/docs/unit_models/reactors/tests/test_egprod_ideal.py

```diff
@@ -0,0 +1,431 @@
+#################################################################################
+# The Institute for the Design of Advanced Energy Systems Integrated Platform
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES).
+#
+# Copyright (c) 2018-2023 by the software owners: The Regents of the
+# University of California, through Lawrence Berkeley National Laboratory,
+# National Technology & Engineering Solutions of Sandia, LLC, Carnegie Mellon
+# University, West Virginia University Research Corporation, et al.
+# All rights reserved.  Please see the files COPYRIGHT.md and LICENSE.md
+# for full copyright and license information.
+#################################################################################
+"""
+Author: Brandon Paul
+"""
+import pytest
+from pyomo.environ import (
+    check_optimal_termination,
+    ConcreteModel,
+    Set,
+    value,
+    Var,
+    units as pyunits,
+)
+from pyomo.util.check_units import assert_units_consistent
+from pyomo.common.unittest import assertStructuredAlmostEqual
+
+from idaes.core import Component
+from idaes.core.util.model_statistics import (
+    degrees_of_freedom,
+    fixed_variables_set,
+    activated_constraints_set,
+)
+from idaes.core.solvers import get_solver
+
+from idaes.models.properties.modular_properties.base.generic_property import (
+    GenericParameterBlock,
+)
+
+from idaes.models.properties.modular_properties.state_definitions import FpcTP
+
+from idaes_examples.notebooks.docs.unit_models.reactors.egprod_ideal import config_dict
+
+from idaes.models.properties.tests.test_harness import PropertyTestHarness
+
+
+# -----------------------------------------------------------------------------
+# Get default solver for testing
+solver = get_solver()
+
+
+def _as_quantity(x):
```

Comment:
I do not think you need this - why not use `from pyomo.environ import as_quantity`?